tests/components/zha/conftest.py

```diff
@@ -7,35 +8,46 @@
 import zigpy
 from zigpy.application import ControllerApplication
 
-from homeassistant import config_entries
 from homeassistant.components.zha.core.const import COMPONENTS, DATA_ZHA, DOMAIN
 from homeassistant.components.zha.core.gateway import ZHAGateway
 from homeassistant.components.zha.core.store import async_get_registry
 from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
 
 from .common import FakeDevice, FakeEndpoint, async_setup_entry
 
+from tests.common import MockConfigEntry
+
 FIXTURE_GRP_ID = 0x1001
 FIXTURE_GRP_NAME = "fixture group"
 
 
 @pytest.fixture(name="config_entry")
-def config_entry_fixture(hass):
+async def config_entry_fixture(hass):
     """Fixture representing a config entry."""
-    config_entry = config_entries.ConfigEntry(
-        1,
-        DOMAIN,
-        "Mock Title",
-        {},
-        "test",
-        config_entries.CONN_CLASS_LOCAL_PUSH,
-        system_options={},
-    )
+    config_entry = MockConfigEntry(domain=DOMAIN)
+    config_entry.add_to_hass(hass)
     return config_entry
 
 
+@pytest.fixture
+async def setup_zha(hass, config_entry):
+    """Load the ZHA component.
+
+    This will init the ZHA component. It loads the component in HA so that
+    we can test the domains that ZHA supports without actually having a zigbee
+    network running.
+    """
+    # this prevents needing an actual radio and zigbee network available
+    with patch("homeassistant.components.zha.async_setup_entry", async_setup_entry):
```

Comment:
We should patch the radio libraries instead. If we avoid patching our own code the tests will be more robust.

We already patch the `ZHAGateway` so it doesn't look too complicated to patch out the libraries where needed.

We want to end up with just setting up the config entry, to load the integration.

```py
entry = MockConfigEntry(...)
entry.add_to_hass(hass)
await hass.config_entries.async_setup(entry.entry_id)
```

Reply:
Let me look into it. I just might have an idea

Reply:
Resolved in https://github.com/home-assistant/home-assistant/pull/31744